### DISCUSSION TITLE
What should I do If I don't have Environments?

### DISCUSSION THREAD

**Discussion by takielias** [serversideup/spin#55] [Q&A] [answered]
As a general user, I don't have the following feature of github in private repo.

![image](https://github.com/serversideup/spin/assets/38932580/39613981-d48a-417d-b272-14ea533daf41)

So, How should I set the 7 secrets for CI/CD? 

I have only Deploy Keys

![image](https://github.com/serversideup/spin/assets/38932580/37591390-e67f-42b8-beea-7b6225c5acb4)

Is it suitable for spin?

**Comment by jaydrogers** [comment, marked as answer]
Hey @takielias,

Deploy keys are used for read-only SSH access to the repository. Try setting **Secrets and Variables** below deploy keys.

The only annoying part is **$MY_VARIABLE** can't be scoped on a per environment basis. This means you'll have to have **$PRODUCTION_MY_VARIABLE**. You'll also have to remove any `environment:` references in the GitHub Actions file.
